Account/FirefoxAccount.swift

```diff
@@ -197,67 +198,52 @@ open class FirefoxAccount {
     }
     
     public class FxAProfile {
-        open var avatarUrl: URL?
-        open var avatarImageData: Data?
         open var displayName: String?
         open let email: String
+        open var avatar: Avatar?
         
         init(email: String, displayName: String?, avatar: String?) {
             self.email = email
             self.displayName = displayName
-            self.avatarImageData = nil
 
             if let avatarStr = avatar {
-                self.avatarUrl = URL(string: avatarStr)
-            } else {
-                self.avatarUrl = nil
-            }            
+                self.avatar = Avatar(url: URL(string: avatarStr))
+            }
         }
         
-        func downloadAvatarImage() {
-            guard let avatarUrl = self.avatarUrl else {
-                return
-            }
+        open class Avatar {
+            open var image: UIImage? = nil
+            open var downloadSucceded: Bool? = false
+            open var url: URL?
             
-            DispatchQueue.global().async {
-                self.avatarImageData = try? Data(contentsOf: avatarUrl)
-                NotificationCenter.default.post(name: NotificationFirefoxAccountProfileChanged, object: self)
+            init(url: URL?) {
+                guard let url = url else {
+                    self.image = UIImage(named: "placeholder-avatar")
+                    return
+                }
+                
+                SDWebImageManager.shared().downloadImage(with: url, options: SDWebImageOptions.continueInBackground, progress: nil) { (image, error, cacheType, success, url) in
+                    guard image != nil else {
+                        self.downloadSucceded = false
```

Comment:
You don't actually use `downloadSucceded` (typo, btw).

I think you intend to use it to allow for retries, to support lazy loading, or something like that.

If so, it's not enough to have `downloadSucceeded`. There are six states you could be in:

1. We have not yet attempted a download.
2. A download has begun.
3. The download failed, and can be retried. (E.g., the connection dropped due to a transient error.)
4.a. The download failed, and should not be retried in this session. (E.g., 404.)
4.b. The download failed, and should never be retried.
5. The download succeeded, and we don't have an image (e.g., it was a malformed file).
6. The download succeeded, and we have an image.